File: src/main/java/com/aquatrack/service/ResidentWaterUsageService.java
package com.aquatrack.service;

import com.aquatrack.dto.resident.ResidentWaterUsageResponseDto;

import java.util.List;

public interface ResidentWaterUsageService {

    // ==========================================
    // Water Usage
    // ==========================================

    /**
     * Retrieves the complete water usage history
     * of the logged-in resident's household.
     *
     * @return list of water usage records
     */
    List<ResidentWaterUsageResponseDto> getMyWaterUsage();

}

File: Aquatrack-frontend/src/services/residentWaterUsageService.js
import api from "../api/axios";

const BASE_URL = "/resident";

// ==========================================
// Resident Water Usage Service
// ==========================================

const residentWaterUsageService = {

    /**
     * Fetch water usage history of the
     * logged-in resident.
     *
     * GET /api/resident/water-usage
     */
    async getMyWaterUsage() {

        const response = await api.get(
            `${BASE_URL}/water-usage`
        );

        return response.data.data;

    },

};

export default residentWaterUsageService;